components/AudioPlayer.js

```diff
@@ -0,0 +1,134 @@
+import React, { useState, useRef, useEffect } from 'react'
+import styles from "../components/AudioPlayer.module.css";
+import utilsStyles from '../styles/utils.module.css';
+import Image from 'next/image';
+import { BsArrowLeftShort } from "react-icons/bs"
+import { BsArrowRightShort } from "react-icons/bs"
+import { FaPlay } from "react-icons/fa"
+import { FaPause } from "react-icons/fa"
+//
+
+const AudioPlayer = () => {
+  // state
+  const [isPlaying, setIsPlaying] = useState(false);
+  const [duration, setDuration] = useState(0);
+  const [currentTime, setCurrentTime] = useState(0);
+
+  // references
+  const audioPlayer = useRef();   // reference our audio component
+  const progressBar = useRef();   // reference our progress bar
+  const animationRef = useRef();  // reference the animation
+
+  useEffect(() => {
+    const seconds = Math.floor(audioPlayer.current.duration);
+    setDuration(seconds);
+    progressBar.current.max = seconds;
+  }, [audioPlayer?.current?.loadedmetadata, audioPlayer?.current?.readyState]);
+
+  const calculateTime = (secs) => {
+    const minutes = Math.floor(secs / 60);
+    const returnedMinutes = minutes < 10 ? `0${minutes}` : `${minutes}`;
+    const seconds = Math.floor(secs % 60);
+    const returnedSeconds = seconds < 10 ? `0${seconds}` : `${seconds}`;
+    return `${returnedMinutes}:${returnedSeconds}`;
+  }
+
+  const togglePlayPause = () => {
+    const prevValue = isPlaying;
+    setIsPlaying(!prevValue);
+    if (!prevValue) {
+      audioPlayer.current.play();
+      animationRef.current = requestAnimationFrame(whilePlaying)
+    } else {
+      audioPlayer.current.pause();
+      cancelAnimationFrame(animationRef.current);
+    }
+  }
+
+  const whilePlaying = () => {
+    progressBar.current.value = audioPlayer.current.currentTime;
+    changePlayerCurrentTime();
+    animationRef.current = requestAnimationFrame(whilePlaying);
+  }
+
+  const changeRange = () => {
+    audioPlayer.current.currentTime = progressBar.current.value;
+    changePlayerCurrentTime();
+  }
+
+  const changePlayerCurrentTime = () => {
+    progressBar.current.style.setProperty('--seek-before-width', `${progressBar.current.value / duration * 100}%`)
+    setCurrentTime(progressBar.current.value);
+  }
```

Comment:
![スクリーンショット 2021-12-12 22 12 37](https://user-images.githubusercontent.com/51708229/145769646-4c94f128-537f-4162-b253-566951a6bf8f.png)
I found the other progress bar here which is not on our mock-up design.
This is because you set `.progressBar::before` and styled the other bar. (Please refer `AudioPlayer.module.css`)
We don't need this, please remove it.

Also, we don't really need `changePlayerCurrentTime` function and we can just add `setCurrentTime(progressBar.current.value);` to `whilePlaying` and `changeRange` function like below.
```suggestion
  const whilePlaying = () => {
    progressBar.current.value = audioPlayer.current.currentTime;
    setCurrentTime(progressBar.current.value);
    animationRef.current = requestAnimationFrame(whilePlaying);
  }

  const changeRange = () => {
    audioPlayer.current.currentTime = progressBar.current.value;
    setCurrentTime(progressBar.current.value);
  }

```